### PR TITLE
Bundle the extension

### LINKED ISSUE #2450: Bundle the PowerShell extension
See https://code.visualstudio.com/api/working-with-extensions/bundling-extension

### PR DESCRIPTION
**Pull request by andschwa**
This uses `esbuild` as outlined in the [VS Code docs](https://code.visualstudio.com/api/working-with-extensions/bundling-extension#using-esbuild) to bundle the PowerShell extension.

Resolves #2450.

### CODE REVIEW

**Comment by andschwa** [comment]
Wow, so this is really annoying. The tooling is absolutely not ready for tests yet.

**Comment by andschwa** [comment]
@rjmholt and @SeeminglyScience please test this (and especially the launch configurations). Everything _seems_ to be working right but it wasn't trivial.

**Comment by andschwa** [comment]
Also filed #3561 as a next step.

File: tsconfig.json [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
-            "DOM"
-        ],
-        "sourceMap": true
-    },
-    "exclude": [
-        "node_modules"
-    ]
+  "compilerOptions": {
+    // NOTE: The TypeScript compiler is only used for building the tests (and
+    // the sources which the tests need). The extension is built with `esbuild`.
```
Comment (rjmholt) [inline tsconfig.json:4]: Does this file configure the typescript compiler rather than `esbuild`?

**Review by rjmholt — APPROVED** [review]

File: tsconfig.json (hunk 1 above)
Comment (andschwa) [reply tsconfig.json:4]: Yes, T(ype)S(cript)config.json. While `esbuild` _supports_ transpiling typescript to javascript, it _does not_ do any sort of type checking etc., and it also will _only_ produce a bundled file. So we still need `tsc` around to type-check the sources and transpile the test code.

**Pull request merged**